pkg/reconciler/tenancy/workspacemounts/workspacemounts_reconcile_updater.go

```diff
@@ -35,102 +36,74 @@ import (
 // we should add second reconciler that will update the status of the workspace so triggering
 // it to be "not ready" if the mount is not ready.
 type workspaceStatusUpdater struct {
-	getMountObject func(ctx context.Context, cluster logicalcluster.Path, ref *tenancyv1alpha1.ObjectReference) (*unstructured.Unstructured, error)
+	getMountObject func(ctx context.Context, cluster logicalcluster.Path, ref tenancyv1alpha1.ObjectReference) (*unstructured.Unstructured, error)
 }
 
 func (r *workspaceStatusUpdater) reconcile(ctx context.Context, workspace *tenancyv1alpha1.Workspace) (reconcileStatus, error) {
-	v, ok := workspace.Annotations[tenancyv1alpha1.ExperimentalWorkspaceMountAnnotationKey]
-	if !ok {
-		// no mount annotation, might be nothing or mount was soft "deleted" by removing the annotation.
-		// Delete the condition.
-		conditions.Delete(workspace, tenancyv1alpha1.MountConditionReady)
+	if workspace.Spec.Mount == nil {
 		return reconcileStatusContinue, nil
 	}
-	mount, err := tenancyv1alpha1.ParseTenancyMountAnnotation(v)
+
+	if !workspace.DeletionTimestamp.IsZero() {
+		return reconcileStatusContinue, nil
+	}
+
+	obj, err := r.getMountObject(ctx, logicalcluster.From(workspace).Path(), workspace.Spec.Mount.Reference)
 	if err != nil {
+		return reconcileStatusStopAndRequeue, err
+	} else if kerrors.IsNotFound(err) {
```

Comment:
This doesn't look right, it would never enter the `else if` branch, no?

Reply:
+1